src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -175,6 +179,106 @@ public void testDBSchema() throws SQLException {
+                if (schemaName.equals(testSchema)) {
+                    hasDashCatalogSchema = true;
+                    assertEquals(catalogName, testCatalog);
+                } else {
+                    assertNull(catalogName);
+                }
+            }
+
+            final MessageFormat atLeastOneFoundFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
+            assertTrue(hasResults, atLeastOneFoundFormat.format(schemaMsgArgs));
+
+            final MessageFormat dashCatalogFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
+            assertTrue(hasDashCatalogSchema, dashCatalogFormat.format(new Object[] {testSchema}));
+        } finally {
+            if (dropDatabase) {
+                connection.createStatement().execute(String.format("DROP DATABASE [%s]", testCatalog));
+            }
+        }
+    }
+
+    /**
+     * Tests that the catalog parameter containing - is escaped by
+     * {@link SQLServerDatabaseMetaData#getSchemas(String catalog, String schemaPattern)}.
+     * @throws SQLException
+     */
+    @Test
+    public void testDBSchemasForDashedCatalogNameWithPattern() throws SQLException {
+        final UUID id = UUID.randomUUID();
+        final String testCatalog = "dash-catalog"+id;
+        final String testSchema = "some-schema"+id;
+        boolean dropDatabase = false;
+
+        try (final Connection dashConn = DriverManager.getConnection(connectionString);
+             final Statement dashStatement = dashConn.createStatement()) {
+
+            connection.createStatement().execute(String.format("CREATE DATABASE [%s]", testCatalog));
+            dropDatabase = true;
+            dashStatement.execute(String.format("USE [%s]", testCatalog));
+            dashStatement.execute(String.format("CREATE SCHEMA [%s]", testSchema));
+
+            final DatabaseMetaData databaseMetaData = connection.getMetaData();
+            final ResultSet rs = databaseMetaData.getSchemas(testCatalog, "some-%");
+
+            final MessageFormat schemaEmptyFormat = new MessageFormat(TestResource.getResource("R_nameEmpty"));
+            final Object[] schemaMsgArgs = {testSchema};
+            final Object[] catalogMsgArgs = {testCatalog};
+
+            boolean hasResults = false;
+            while (rs.next()) {
+                hasResults = true;
+                final String schemaName = rs.getString(1);
+                final String catalogName = rs.getString(2);
+                assertTrue(!StringUtils.isEmpty(schemaName), schemaEmptyFormat.format(schemaMsgArgs));
+                assertTrue(!StringUtils.isEmpty(catalogName), schemaEmptyFormat.format(catalogMsgArgs));
+                assertEquals(schemaName, schemaMsgArgs[0]);
+                assertEquals(catalogName, catalogMsgArgs[0]);
+            }
+
+            final MessageFormat atLeastOneFoundFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
```

Comment:
Everything else looks okay, there's just too many `final` objects here in both new methods, which don't really have a purpose. Let's keep it consistent with other tests.

Reply:
Removed all instances of `final`.